controllers/hetznerbaremetalhost_controller.go

```diff
@@ -97,6 +97,18 @@ func (r *HetznerBareMetalHostReconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// Remove permanent error, if the corresponding annotation was removed by the user.
+	requeue := removePermanentErrorIfAnnotationIsGone(bmHost)
+	if requeue {
```

Comment:
this part looks a bit weird. Why do we update the host after requeue is true?
Maybe we can rename the variables and functions to improve the flow. "removed" instead of "requeue" for example would make it a lot better IMO

Reply:
@janiskemper changed in 5bf44503 - better now?